packages/vite/src/node/server/environment.ts

```diff
@@ -118,16 +118,16 @@ export class DevEnvironment extends BaseEnvironment {
       this.pluginContainer!.resolveId(url, undefined),
     )
 
-    this.hot = setup.hot || createNoopHotChannel()
-    this.watcher = setup.watcher
+    this.hot = setup?.hot ?? createNoopHotChannel()
```

Comment:
The `??` doesn't really make sense to me in this context - just adds unnecessary compiled code. Even when `hot` was `false`, `||` worked just fine